### DISCUSSION TITLE
Issues running export vtk tutorial

### DISCUSSION THREAD

**Discussion by savendano-gw** [modflowpy/flopy#1886] [Q&A] [answered]
Hello,

I am stuck on the line 380:  notebook_utils.run(workspace)

in the export vtk tutorial

I keep getting the error: 
AttributeError: module 'notebook_utils' has no attribute 'run'

I have googled this issue to no avail. Are there any alternatives to this line?

Apologies if this is not the correct place for this question.

Thank you!

**Comment by wpbonelli** [comment]
Hi @savendano-gw, most flopy notebooks are not dependency-free (yet) &mdash; the VTK export notebook depends on the `notebook_utils` module [here](https://github.com/modflowpy/flopy/blob/develop/.docs/common/notebook_utils.py), imported [via sys.path manipulation](https://github.com/modflowpy/flopy/blob/8c330a7890fe732c5489ce6a2d5184a7b2fb70ee/.docs/Notebooks/export_vtk_tutorial.py#L40). You can download this file and update the path to import it. Note you will also need the example data in `examples/data`.

Work is underway to make notebooks self-contained (see https://github.com/modflowpy/flopy/issues/1872), this could be folded into that effort.

**Comment by savendano-gw** [comment]
Thank you!

**Reply by wpbonelli** [reply]
Sure thing! Were you able to successfully run the notebook?

**Reply by savendano-gw** [reply]
Unfortunately not; I am running into this error:
File ~\Anaconda3\envs\spyder\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\savendan\documents\flopy_tutorial\export_vtk_tutorial.py:381
    notebook_utils.run(workspace)

  File ~\Documents\flopy_tutorial\notebook_utils.py:465 in run
    raise ValueError("Failed to run.")

ValueError: Failed to run.

**Reply by wpbonelli** [reply]
If you edit line 368 of `notebook_utils.py` and switch `silent=True` to `silent=False`, you should see logging output from the model run and we can find the problem. Alternatively you can remove `if success:` guarding the next 2 lines to print captured logs.

It's really not helpful for `notebook_utils.py` to suppress output on failure, sorry for that..

```shell
> git blame -c -L 368 .docs/common/notebook_utils.py | head -n 6
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       368)    success, buff = sim.run_simulation(silent=True, report=True)
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       369)    if success:
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       370)        for line in buff:
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       371)            print(line)
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       372)    else:
fb3504bad       ( w-bonelli     2023-02-24 16:34:51 -0500       373)        raise ValueError("Failed to run.")
```

**Reply by savendano-gw** [reply]
ok I edited silent=True to silent=False; this is what I see:

  MODFLOW-2005     
    U.S. GEOLOGICAL SURVEY MODULAR FINITE-DIFFERENCE GROUND-WATER FLOW MODEL
                             Version 1.12.00 2/3/2017                        

 Using NAME file: mp7p2b_mp.mpsim 
 Run start date and time (yyyy/mm/dd hh:mm:ss): 2023/08/02 14:39:24


 CANNOT OPEN   ON UNIT   0 BECAUSE UNIT IS ALREADY BEING USED

**Reply by wpbonelli** [reply, marked as answer]
This notebook should only run mf6 and mp7 models, not mf2005. I suspect `mf2005` is available on your path, but `mp7` is not. Flopy silently defaults to mf2005 if the specified executable cannot be found (I think [this should change](1896))

You can run `get-modflow : --subset mp7` to install MODPATH 7. It will prompt you to select an install location. Wherever you choose, it will add it to the path. (the bin or scripts folder of your conda environment is a good place, as it keeps things isolated.)

**Reply by savendano-gw** [reply]
Its working! Thank you so much!
